### PR TITLE
AC2 PSE 5 - Rotate Linked List

### PR DESCRIPTION
**Pull request by kyra-at-ada**
Summary of Changes:
- Archive PSEs not being used for AC2
- Renumber PSEs according to AC2 schedule
- Add Rotate Linked List PSE
    - Question Intro
    - Clarifying Questions
    - Write Unit Tests
    - Write Pseudocode
    - Implement the question (checkpoint)
    - Two possible solutions

Additional Notes:
I ultimately decided to just show them how to represent a linked list in the hint for writing unit tests instead of writing out an example nominal case. There is also a note in the question that the `==` operator for a node will test the equality of the entire list (given the head nodes of two lists). I thought it might be a good exercise to see if they can put the components together, but can easily change it to include an entire example test case depending on y'alls thoughts.

For Review:
- Please preview the lesson in Learn to ensure there are no errors with publishing
- Test sample solutions against checkpoint question
- Check for test coverage in checkpoint
- Verify accuracy of test cases and examples
- Check formatting is consistent with other PSEs
- Spelling/grammar

### CODE REVIEW

File: 05/rotate-linked-list.checkpoint.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
+# Implement Rotate Linked List
+
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: code-snippet
+* language: python3.6
+* id: 48651a9f-90c7-490a-90ab-0ef771f47170
+* title: Merging Sorted Lists
```
Comment (char-adadev) [inline 05/rotate-linked-list.checkpoint.md:9]:
```suggestion
* title: Rotate Linked List
```

File: 05/rotate-linked-list.instructor.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    # Move the last element to the front k times
+    for _ in range(k):
+        head = pop_to_front(head)
+        
+    return head
+```
+
+# O(n) solution
+```python
+def rotate(head, k):
```
Comment (char-adadev) [inline 05/rotate-linked-list.instructor.md:47]:
```suggestion
def rotate_list(head, k):
```

**Review by char-adadev — COMMENTED** [review]
Thank you for working on this! I had a couple of small suggestions. Overall, everything looks good to me. The learn preview ran without issue and the solutions presented in the Instructor notes worked fine, with a small update (one of the suggestions).

**Review by kelsey-steven-ada — COMMENTED** [review]
Thank you for the time and effort on the Accelerate curriculum, this is looking great overall!

I was looking at sequencing, it looks like the PSE order is a little different than what was in the changelogs. I’m good with “merging sorted lists” moving forward since it’s a bit easier and this puts it in unit 2 rather than 3. Taking a look at the roundtables, we don’t cover BSTs until week 4 of Unit 3 and Graphs until week 5. If the remaining PSEs will be trees and graphs, we might need to shift the relevant class sessions forward a week and make “Best time to Buy/Sell Stock” PSE 08 so folks can be prepared for the remaining PSEs.

This is fully outside of the PR feedback, but seeing the new lesson reminded me – around the Big O explanations, is it possible, and has there been discussion about using the larger text box option that we see in lesson takeaways? It’s a minor pain point for grading; when students give a great breakdown of their complexity, I generally need to copy/paste the response into a file so I can see it all.

**Comment by kyra-at-ada** [comment]
>
Thank you for the time and effort on the Accelerate curriculum, this is looking great overall!
> 
> I was looking at sequencing, it looks like the PSE order is a little different than what was in the changelogs. I’m good with “merging sorted lists” moving forward since it’s a bit easier and this puts it in unit 2 rather than 3. Taking a look at the roundtables, we don’t cover BSTs until week 4 of Unit 3 and Graphs until week 5. If the remaining PSEs will be trees and graphs, we might need to shift the relevant class sessions forward a week and make “Best time to Buy/Sell Stock” PSE 08 so folks can be prepared for the remaining PSEs.
> 
> This is fully outside of the PR feedback, but seeing the new lesson reminded me – around the Big O explanations, is it possible, and has there been discussion about using the larger text box option that we see in lesson takeaways? It’s a minor pain point for grading; when students give a great breakdown of their complexity, I generally need to copy/paste the response into a file so I can see it all.

@kelsey-steven-ada 

I can definitely change that for all of the PSEs that haven't yet been released and update for other PSEs going forward. 

On ordering, I think I got a bit confused matching the numbering in the repo with the changelog. I can redo that. 

That would make the ordering
PSE 01 - Hamming
PSE 02 - Palindrome
PSE 03 - Pairs with Given Sum
PSE 04 - Best time to buy/sell stock
PSE 05 - Rotate a Linked List
PSE 06 - Reshape a Matrix
PSE 07 - Kth Missing Positive Number
PSE 08 - Merge Sorted Lists 
PSE 09 - BSTs 
PSE 10 - Graphs

**Comment by kelsey-steven-ada** [comment]
@kyra-at-ada Looking at the order in core and the problems themselves I'd like to swap "Best time to buy/sell stock" and "Merge Sorted Lists", so that "Merge Sorted Lists" is PSE 04 and "Best time to buy/sell stock" is PSE 08. Everything else looks good to me 😄

**Pull request merged**